compiler/x/i386/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3643,6 +3645,36 @@ TR::Register *OMR::X86::I386::TreeEvaluator::lternaryEvaluator(TR::Node *node, T
                                    falseReg->getRegisterPair()->getHighOrder(), cg);
          }
       }
+   else if (condOp.isCompareForOrder() && condition->getFirstChild()->getOpCode().isIntegerOrAddress() && !longCompare)
```

Comment:
Similarly, it may be worthy bring `!longCompare` as the first condition, so that the rest checks can be skipped when `!longCompare` is false.

Reply:
addressed.